plugins/parsers/prometheusremotewrite/parser.go

```diff
@@ -0,0 +1,90 @@
+package prometheusremotewrite
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+type Parser struct {
+	DefaultTags map[string]string
+}
+
+func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
+	var err error
+	var metrics []telegraf.Metric
+	var req prompb.WriteRequest
+
+	if err := proto.Unmarshal(buf, &req); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal request body: %s", err)
+	}
+
+	now := time.Now()
+
+	for _, ts := range req.Timeseries {
+		tags := map[string]string{}
+		for key, value := range p.DefaultTags {
+			tags[key] = value
+		}
+
+		for _, l := range ts.Labels {
+			tags[l.Name] = l.Value
+		}
+
+		metricName := tags[model.MetricNameLabel]
+		if metricName == "" {
+			return nil, fmt.Errorf("metric name %q not found in tag-set or empty", model.MetricNameLabel)
+		}
+		delete(tags, model.MetricNameLabel)
+
+		for _, s := range ts.Samples {
+			fields := make(map[string]interface{})
+			if !math.IsNaN(s.Value) {
+				fields[metricName] = s.Value
+			}
+			// converting to telegraf metric
+			if len(fields) > 0 {
+				var t time.Time
+				if s.Timestamp > 0 {
+					t = time.Unix(0, s.Timestamp*1000000)
+				} else {
+					t = now
+				}
+				metric, err := metric.New("prometheusremotewrite", tags, fields, t)
```

Comment:
so the metric name from above is being used as the fieldname, it's a similar thing to whats in our prometheus parser so the telegraf metric name is hardcoded and the metric name from prometheus is used within an fieldname. Otherwise the output would become: 

`scrape_samples_scraped,instance=localhost:9090,job=prometheus scrape_samples_scraped=390 1615920053859000000`

do you think I should change it to make it clearer that it's not becoming the telegraf metric name? Could add a comment in the code or similar.

Reply:
seems like that should maybe be a configuration option. I know it's possible to do it with the rename plugin, but this seems relevant. a name prefix might also be helpful.